Utils/utils.ts

```diff
@@ -57,17 +57,29 @@ export class utils {
         }
     }
 
-    async click(selector: string | Locator, nextSelector: string, timeout?: number): Promise<void> {
+    async click(selector: string | Locator, nextSelector: string | Locator, timeout?: number): Promise<void> {
         try {
             let locator = typeof selector === 'string' ? await this.locator(selector) : selector;
             await locator.click();
-            await this.waitForSelector(nextSelector, { timeout: timeout })
+            
+            if (typeof nextSelector === 'string')
+            {
+                await this.waitForSelector(nextSelector, { timeout: timeout })
+            }
+            else if(!await nextSelector.isVisible())
+            {
+                throw Error('next locator is not visible after clicking the selector')
+            }
         } catch (error) {
             console.log('failed to click the selector ' + selector + '\nwith error : ' + error);
```

Comment:
please use the throw instead of console logs (remove console logs), combine the error message description with the error:

ex: throw new Error('failed to click the selector ' + selector + '\nwith error : ' + error)